### PR TITLE
Added docs to CupertinoAlertDialog to explain title, content, and but…

### PR DESCRIPTION
**Pull request by matthew-carroll**
#18128

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/dialog.dart [hunk 1]
```diff
@@ -106,16 +106,25 @@ class CupertinoDialog extends StatelessWidget {
...
 /// An alert dialog informs the user about situations that require
-/// acknowledgement. An alert dialog has an optional title and an optional list
-/// of actions. The title is displayed above the content and the actions are
-/// displayed below the content.
+/// acknowledgement. An alert dialog has an optional title, optional content,
+/// and an optional list of actions. The title is displayed above the content
+/// and the actions are displayed below the content.
+///
+/// This dialog styles its title and content to match the standard iOS title
+/// and content dialog text style. These default styles can be overridden by
```
Comment (xster) [inline packages/flutter/lib/src/cupertino/dialog.dart]: iOS has a title and a message. We can just say `This dialog styles its title and content (typically a message) to match the standard iOS title and message dialog text style`

File: packages/flutter/lib/src/cupertino/dialog.dart [hunk 2]
```diff
@@ -106,16 +106,25 @@ class CupertinoDialog extends StatelessWidget {
...
+/// explicitly defining [TextStyle]s for [Text] widgets that are part of the
+/// title or content.
+///
+/// To display action buttons that look like standard iOS dialog buttons,
+/// provide [CupertinoDialogAction]s for the 'actions' given to this dialog.
```
Comment (xster) [inline packages/flutter/lib/src/cupertino/dialog.dart]: nit: backticks for parameters that aren't also fields. Square brackets for parameters that are also fields. Normal quotes for things that aren't code.

**Review by xster — APPROVED** [review]
LGTM

Thanks!

**Comment by matthew-carroll** [comment]
@xster I made both corrections.  Let me know if there are any other issues.  Thanks.

**Comment by xster** [comment]
ship it

**Pull request merged**